test-apps/sediment/src/main/java/com/twosigma/waiter/sediment/Servlet.java

```diff
@@ -0,0 +1,225 @@
+                LOGGER.info("total bytes read = " + totalBytesRead);
+                break;
+            }
+            totalBytesRead += bytesRead;
+        }
+
+        return totalBytesRead;
+    }
+
+    @Override
+    protected void service(final HttpServletRequest request, final HttpServletResponse response)
+        throws IOException, ServletException {
+
+        final String correlationId = retrieveCorrelationId(request);
+        LOGGER.info(correlationId + " received " + request.getProtocol() + " " + request.getMethod() +
+            " request at " + request.getRequestURI());
+        response.setHeader("x-cid", correlationId);
+        response.setHeader("server", "sediment:" + HttpConfiguration.SERVER_VERSION);
+
+        final Request jettyRequest = (Request) request;
+        final Response jettyResponse = (Response) response;
+
+        switch (request.getRequestURI()) {
+            case "/status": {
+                handleStatusRequest(jettyRequest, jettyResponse);
+                break;
+            }
+            case "/trailers": {
+                handleTrailersRequest(jettyRequest, jettyResponse);
+                break;
+            }
+            default: {
+                LOGGER.severe(correlationId + " unsupported endpoint: " + request.getRequestURI());
+                super.service(request, response);
+                break;
+            }
+        }
+        LOGGER.info(correlationId + " done processing request");
+    }
+
+    private void handleStatusRequest(final Request request, final Response response) throws IOException {
+        final String contentParam = request.getParameter("content");
+        final String responseContent = contentParam == null ? "OK" : contentParam;
+        response.setContentType("text/plain");
+        response.getWriter().write(responseContent);
+    }
+
+    private void handleTrailersRequest(final Request request, final Response response) throws IOException {
+        final Map<String, Object> infoMap = populateRequestInfo(request);
+        final String responseBody = new Gson().toJson(infoMap);
+        final byte[] outputBytes = responseBody.getBytes();
+
+        final HttpFields trailer = createResponseTrailers(request);
+        if (trailer != null) {
+            response.setTrailers(() -> {
+                sleepBasedOnRequestHeader(
+                    request, "x-sediment-sleep-before-response-trailer-ms", "response trailers");
+                return trailer;
+            });
+        }
+
+        final String responseStatusString = request.getHeader("x-sediment-response-status");
+        if (responseStatusString == null) {
+            response.setStatus(200);
+        } else {
+            response.setStatus(Integer.parseInt(responseStatusString));
+        }
+
+        final ByteArrayInputStream respInputStream = new ByteArrayInputStream(outputBytes);
+        final ServletOutputStream respOutputStream = response.getOutputStream();
```

Comment:
I use the output stream instead of the writer to mimic chunked encoding response below while working with byte arrays.